src/tax-lot/tax-lot.repo.ts

```diff
@@ -0,0 +1,175 @@
+import { Inject } from "@nestjs/common";
+import { DB, DbType } from "src/global/providers/db.provider";
+import { eq, sql } from "drizzle-orm";
+import { DataRetrievalException, ResourceNotFoundException } from "src/error";
+import {
+  SelectTaxLot,
+  SelectTaxLotNested,
+  SelectTaxLotSpatial,
+} from "src/schema/tax-lot";
+import {
+  taxLot,
+  zoningDistrict,
+  zoningDistrictClass,
+  zoningDistrictZoningDistrictClass,
+} from "src/schema";
+import { MultiPolygon } from "geojson";
+
+export class TaxLotRepo {
+  constructor(
+    @Inject(DB)
+    private readonly db: DbType,
+  ) {}
+
+  #checkTaxLotByBbl = this.db.query.taxLot
+    .findFirst({
+      columns: {
+        bbl: true,
+      },
+      where: (taxLot, { eq, sql }) => eq(taxLot.bbl, sql.placeholder("bbl")),
+    })
+    .prepare("checkTaxLotByBbl");
+
+  async taxLotByBbl(bbl: string) {
+    let result: SelectTaxLotNested | undefined;
+    try {
+      result = await this.db.query.taxLot.findFirst({
+        columns: {
+          boroughId: false,
+          landUseId: false,
+          wgs84: false,
+          liFt: false,
+        },
+        where: (taxLot, { eq }) => eq(taxLot.bbl, bbl),
+        with: {
+          borough: true,
+          landUse: true,
+        },
+      });
+    } catch {
+      throw DataRetrievalException;
+    }
```

Comment:
The suggesting below (moving the undefined check to the service) will also focus this function. In turn, we can simplify its return. 

*Code segment is only for illustrative purposes. Please verify functionality in full coding environment*
```suggestion
    try {
      return await this.db.query.taxLot.findFirst({
        columns: {
          boroughId: false,
          landUseId: false,
          wgs84: false,
          liFt: false,
        },
        where: (taxLot, { eq }) => eq(taxLot.bbl, bbl),
        with: {
          borough: true,
          landUse: true,
        },
      });
    } catch {
      throw DataRetrievalException;
    }
```